Lib/unittest/result.py

```diff
@@ -191,7 +191,8 @@ def _exc_info_to_string(self, err, test):
             capture_locals=self.tb_locals, compact=True)
         from _colorize import can_colorize
 
-        msgLines = list(tb_e.format(colorize=can_colorize()))
+        file = self.stream if hasattr(self, "stream") else None
```

Comment:
```suggestion
        file = getattr(self, "stream", None)
```

Reply:
I think that it should not colorize if there is no stream.
```suggestion
        colorize = hasattr(self, "stream") and can_colorize(self.stream)
```